src/main/java/com/keniu/models/Book.java

```diff
@@ -0,0 +1,44 @@
+package com.keniu.models;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.validation.constraints.NotNull;
+import java.math.BigDecimal;
+
+/**
+ * Represents a book in the book shop application.
+ */
+@Entity
+public class Book {
+
+    /** The unique identifier for the book. */
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    /** The title of the book. */
+    @NotNull
```

Comment:
For string fields, it usually makes more sense to use @NotBlank instead of @NotNull since @NotBlank checks that the string is not null and also not empty. However, keep in mind that validation annotations like @NotNull or @NotBlank only enforce constraints in the application layer (in-memory) and do not apply at the database level. To ensure database-level constraints, it's a good practice to include @Column(nullable = false) as well, which enforces NOT NULL in the database

Reply:
Thanks for information! Changes have been made.